homeassistant/components/ihc/ihcdevice.py

```diff
@@ -41,21 +65,52 @@ def name(self):
         """Return the device name."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"ihc{self.controller_id}{self.ihc_id}"
+
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
-        if not self.info:
+        if not self.hass.data[DOMAIN][self.controller_id][CONF_INFO]:
             return {}
-        return {
+        attributes = {
             "ihc_id": self.ihc_id,
             "ihc_name": self.ihc_name,
             "ihc_note": self.ihc_note,
             "ihc_position": self.ihc_position,
         }
+        multicontroller: bool = len(self.hass.data[DOMAIN]) > 1
+        # We only want to show the controller id if we have more than one
+        if multicontroller:
+            attributes["ihc_controller"] = self.controller_id
```

Comment:
Will this be used for automations?

Reply:
Yes - today the controller index in the yaml file is the controller id in service functions. The controller serial will be the controller id later